src/main/java/com/cvsgo/service/ReviewService.java

```diff
@@ -0,0 +1,52 @@
+package com.cvsgo.service;
+
+import com.cvsgo.dto.review.CreateReviewRequestDto;
+import com.cvsgo.entity.Product;
+import com.cvsgo.entity.Review;
+import com.cvsgo.entity.User;
+import com.cvsgo.repository.ProductRepository;
+import com.cvsgo.repository.ReviewRepository;
+import com.cvsgo.util.FileConstants;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.io.IOException;
+import java.util.List;
+
+import static com.cvsgo.exception.ExceptionConstants.NOT_FOUND_PRODUCT;
+
+@Service
+@RequiredArgsConstructor
+public class ReviewService {
+
+    private final FileUploadService fileUploadService;
+
+    private final ReviewRepository reviewRepository;
+
+    private final ProductRepository productRepository;
+
+    /**
+     * 리뷰를 추가합니다.
+     * @param user 리뷰를 작성한 사용자
+     * @param productId 상품 ID
+     * @param request 사용자가 작성한 리뷰 정보
+     * @throws IOException 파일 접근에 실패한 경우
+     */
+    @Transactional
```

Comment:
```suggestion
    @Transactional(rollbackFor = Exception.class)
```
기본 `@Transaction`은 Unchecked Exception과 Error에만 roll back을 수행하고,
Checked Exception에는 roll back을 수행하지 않아서 `rollbackFor = Exception.class` 를 추가하면 좋을 것 같아요

제가 이번에 상품 좋아요 API 개발하면서 작성한 블로그 링크 첨부합니다🫠
[[Spring Boot] @Transactional(rollbackFor=Exception.class)](https://chaewsscode.tistory.com/180)